### PR TITLE
Docs: allow offline TSC votes

### PR DESCRIPTION
**Pull request by not-an-aardvark**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Documentation update

**What changes did you make? (Give an overview)**

This updates the governance documentation to allow for offline votes on a TSC agenda item if an item is unable to be discussed during a regularly-scheduled TSC meeting.

This has the following goals:

* Formalize the policy of ":+1:s from a majority of TSC members" that we've already been informally following anyway
* Allow time-sensitive TSC agenda items to be resolved when meetings don't reach quorum
* Preserve the ability for a TSC member to object to a proposal and discuss their objections in a meeting

The requirement for offline votes that "the TSC was unable to discuss [the issue] during a regularly-scheduled meeting" is put in place to ensure that TSC members have the opportunity to object to a proposal if they disagree with it. If a TSC member is unable to attend a meeting, then typically they might not be able to voice an objection to an issue anyway (since the issue might get approved by a majority of TSC members in their absence).

**Is there anything you'd like reviewers to focus on?**

Nothing in particular, but feel free to give feedback on the general approach and whether you think this is a good idea.

### CODE REVIEW

**Comment by not-an-aardvark** [comment]
Thanks for the feedback, these sound like good ideas.

> * The request for offline voting must be explicitly stated in the TSC meeting, and must be unanimously agreed upon by all present in the meeting. (Basically, this is a particular type of "resolution" for an issue, and cannot be done implicitly.)

This seems like a good solution for the case where an objecting TSC member is present at the meeting time but quorum is not achieved. Do you envision this working in the case where there isn't enough time to discuss all the agenda items? If there is no more time and people have to leave, it might be difficult to get unanimous agreement about an offline vote. (One potential solution would be to only allow offline votes in cases where a meeting was cancelled due to lack of quorum, although there could be some edge cases about the distinction between the two, e.g. if someone needs to leave close to the end of a meeting and then there is no longer quorum.)

> * If a TSC member objects, either in the original meeting or later on in the issue, then the matter must be brought back to a later TSC meeting (unless the objection can be resolved through discussion on the issue and the objection is withdrawn).
>    * Not sure if your original suggestion includes both within the meeting, as well as in the issue if someone who wasn't at the meeting objects later/outside the meeting.

I agree that if someone objects in a meeting or on an issue before an issue is accepted, then the issue needs to be brought back to a TSC meeting or resolved on the issue.

I'm not sure if this is what you meant, but I don't think it should be possible to object after an issue is already accepted as a result of an offline vote. This could cause confusing messaging about project direction, and some actions taken as a result of an accepted issue might be irreversible. Instead, we would ideally be able to allow time for objections before considering the issue to be accepted.

**Comment by platinumazure** [comment]
> I'm not sure if this is what you meant, but I don't think it should be possible in general to object after an issue is already accepted as a result of an offline vote, and have the status immediately changed as a result of the objection. This seems like it could cause confusing messaging about project direction, and some actions taken as a result of an accepted issue might be irreversible. Instead, we would ideally be able to allow time for objections before considering the issue to be accepted.

Agreed. To that end, maybe we should stipulate that an offline vote requires some minimum amount of time for objections, e.g., 48-72 hours.

> If there is no more time and people have to leave, it might be difficult to get unanimous agreement about an offline vote.

Agreed, my suggestion would be to add time (last 5 minutes) for anyone to request offline vote for time-sensitive issues.

Another possibility would be to allow someone to request a sort of provisional offline vote situation at the beginning of a meeting, or between two (other) topics, with the intent of coming to a resolution with quorum later in the meeting but supplying the offline vote as a fallback. But that might be a bit cumbersome or complicated.

Basically, my vision is that someone could interject with a request for offline vote on any issue more or less anytime (but ideally not in the middle of another issue), and that's a straight up/down vote on whether the issue is time-sensitive or important enough to potentially handle offline. I don't think there needs to be a set formula for that. I'm open to ideas on how to implement; the only thing that matters to me is unanimity in the meeting on allowing an offline vote.

**Comment by platinumazure** [comment]
@not-an-aardvark Thanks, I've reviewed the latest set of changes and they look good so far.

I'd still like for us to figure something out about how to avoid the situation where one person might "request an offline vote" and it's unopposed. At minimum I would suggest that this maneuver can only be done during a TSC meeting in the first place. My preference would be that we have quorum at the time of the request as well, but that might not be feasible as we've already discussed.

**Comment by not-an-aardvark** [comment]
> I've reviewed the latest set of changes and they look good so far.

I'm a bit confused because I haven't yet made any changes to this PR since our discussion. I agree that some changes are needed to address the things you've brought up.

**Comment by platinumazure** [comment]
> I'm a bit confused because I haven't yet made any changes to this PR since our discussion.

Right you are, I thought I had seen a commit since the discussion but I see now I was imagining things. Sorry, please disregard my last comment.

**Comment by nzakas** [comment]
@platinumazure @not-an-aardvark how do we want to move this forward?

**Comment by not-an-aardvark** [comment]
I'll close this for now. Hopefully the "non-attending TSC" status will reduce the number of cancelled meetings.
